### PR TITLE
Feature/control bounce

### PR DESCRIPTION
**Pull request by KtorZ**

### CODE REVIEW

**Review by Soreine — COMMENTED** [review]
Waiting for second iteration :p

**Comment by Soreine** [comment]
There's something weird with the distance I get when I play. They are statistically strange. Here is the log of the frame difference between the bounce and the input:
```
Bouncer.js:68 distance 0
Bouncer.js:68 distance 150
Bouncer.js:68 distance 74
Bouncer.js:68 distance 0
Bouncer.js:68 distance 42
Bouncer.js:68 distance 0
Bouncer.js:68 distance 150
Bouncer.js:68 distance 74
Bouncer.js:68 distance 0
Bouncer.js:68 distance 42
Bouncer.js:68 distance 0
Bouncer.js:68 distance 150
Bouncer.js:68 distance 74
Bouncer.js:68 distance 36
Bouncer.js:68 distance 0
Bouncer.js:68 distance 150
Bouncer.js:68 distance 74
Bouncer.js:68 distance 0
Bouncer.js:68 distance 42
Bouncer.js:68 distance 0
Bouncer.js:68 distance 150
Bouncer.js:68 distance 74
Bouncer.js:68 distance 36
Bouncer.js:68 distance 11
```

**Comment by Soreine** [comment]
OK i fixed it, the input being not debounced, it was being registered again after the bounce resolution.
The distance progression we observed was the number of frames between successive bounces, diminishing as the disc was slowing. :)

File: src/js/game/states/play.js [hunk 1]
```diff
@@ -84,25 +84,23 @@ play.create = function create() {
...
 
     this.bouncer.update(game.input.keyboard.isDown(Phaser.Keyboard.SPACEBAR));
 
-    game.physics.arcade.collide(disc, water, () => {
+    const didCollide = game.physics.arcade.collide(disc, water);
+    if (didCollide) {
         makeSplash(game, disc);
         this.bouncer.bounce();
-    });
+    }
```
Comment (KtorZ) [inline src/js/game/states/play.js]: Why removing them from the callback ? 
Is there any benefits in terms of speed / precision ?

File: src/js/game/states/play.js [hunk 2]
```diff
@@ -90,20 +90,22 @@ play.create = function create() {
...
+        const didCollide = game.physics.arcade.collide(disc, water);
+        if (didCollide) {
+            makeSplash(game, disc);
+        }
+        this.bouncer.update(game.input.keyboard.isDown(Phaser.Keyboard.SPACEBAR), didCollide);
+
+        // The disc was too slow and killed
+        if (!disc.alive) {
+            // Reset the game later
+            setTimeout(() => {
```
Comment (KtorZ) [inline src/js/game/states/play.js:119]: wasn't the `restartTimeout` solving the issue of the `restart()` being triggered multiple times?

File: src/js/game/states/play.js (hunk 1 above)
Comment (Soreine) [reply src/js/game/states/play.js]: well the callback is synchronous, so yeah, it allows to do branch on `collide = false` too

File: src/js/game/states/play.js (hunk 2 above)
Comment (Soreine) [reply src/js/game/states/play.js:119]: It is, but I find this cleaner, and it does not use any additional reference

**Pull request merged**